src/Cache/Psr16CacheClient.php

```diff
@@ -111,7 +111,7 @@ public function getLastError(bool $clear_error = true): ?CacheException
     /**
      * @inheritDoc
      */
-    public function get($key, $default = null)
+    public function get(string $key, mixed $default = null): mixed
```

Comment:
This change will break things for older psr-16 cache versions, and is not needed for newer versions.

Reply:
See https://wiki.php.net/rfc/covariant-returns-and-contravariant-parameters.

Reply:
Our PHP 8 tests were failing because the declaration wasn't compatible: https://github.com/momentohq/client-sdk-php/actions/runs/7833550796/job/21374774900. It seems like fixing it for 7 breaks it for 8 and vice versa. The tests are indicating it is needed for PHP 8.0. Any advice on making this mutually compatible would definitely be appreciated :-)

The larger issue that motivated experimenting with removing `composer.lock` was also due to the PSR-16 interface, as we couldn't get composer to calculate a set of dependencies that is valid for both PHP 7 and PHP 8.x.

Reply:
Ah, sorry, I missed your previous comment. Thanks for the link, I'll check it out!

Reply:
Probably the issue you have is that supporting PHP 7.4 caused you a pain in the ass because it doesn't have the mixed type.

Reply:
You could do some jank and conditionally define the class depending on PHP version. Demo:

1. https://3v4l.org/2sCRB
2. https://3v4l.org/n72t8

You can see that PHP 8 works fine with both the interface with and without types, but not PHP 7.

Reply:
Thanks, Graham, I appreciate the insight.